src/test/lib/storage/pos_lists/entire_chunk_pos_list_test.cpp

```diff
@@ -58,7 +58,41 @@ TEST_F(EntireChunkPosListTest, AddAfterMatchedAllTest) {
   // Newly added rows are not in the position list
   EXPECT_EQ(entire_chunk_pos_list->size(), 3);
 
-  // TODO(XPERIANER): Maybe add a better check than just size, cause the returned iterators should also handle
+  // TODO(anyone): Maybe add a better check than just size, cause the returned iterators should also handle.
   // this case, which we right now don't check.
 }
+
+TEST_F(EntireChunkPosListTest, InsertDoesNotAffectIterators) {
+  // This checks that the EntireChunkPosList does not change its iterators after rows were added to the table. These
+  // added rows should not be contained in the PosList.
+
+  const auto table = Table::create_dummy_table({{"a", DataType::Int, false}});
+  table->append({int32_t{1}});
+  table->append({int32_t{2}});
+  table->append({int32_t{3}});
+
+  EXPECT_EQ(table->chunk_count(), 1);
+  EXPECT_EQ(table->row_count(), 3);
+
+  const auto entire_chunk_pos_list = std::make_shared<const EntireChunkPosList>(ChunkID{0}, ChunkOffset{3});
```

Comment:
The list's iterators shouldn't change when we change the table's content. That's also kind of what I intended to test here

Reply:
There is no connection between these two data structures. So why should something change?

I am not super happy with the test, but we discussed it yesterday and it might be of help, once the list is changed and actually somewhat aware of the table.

Hence, can you add a comment like `// Currently, the EjtirePosList is not linked to the table. We thus don't expect changes when table change. Nonetheless, this test ensures the behaviour remains the same when link is connected/aware of table.` or so?

Reply:
See 84b2372